src/optimalcontrol/src/ConstraintsGroup.cpp

```diff
@@ -159,6 +185,26 @@ namespace optimalcontrol {
                 reportError("ConstraintsGroup", "removeConstraint", errorMsg.str().c_str());
                 return false;
             }
+
+            bool found = false;
+            std::vector<TimedConstraint_ptr>::iterator it = m_pimpl->orderedIntervals.begin();
+            while ((!found) && (it != m_pimpl->orderedIntervals.end())){
+                if (it->get()->constraint->name() == name){
+                    m_pimpl->orderedIntervals.erase(it);
+                    found = true;
+                } else{
+                    ++it;
+                }
+            }
+
+            if(!found){
```

Comment:
space between if and bracket, and between closing bracket and opening curly bracket